src/watchmaker/workers/yum.py

```diff
@@ -54,40 +45,25 @@ def _get_amazon_el_version(self, version):
 
     def get_dist_info(self):
         """Validate the Linux distro and return info about the distribution."""
-        dist = None
```

Comment:
These three assignments for `dist`, `version`, and `el_version` should probably be kept here at the top, to ensure they are set and have values (`None`). Otherwise, they are set in `try` or `if` blocks, which means they may not be set at all if there is an error or an unexpected condition.

Reply:
The `except` should abort the overall execution, though. (and the `version` and `el_version` are still kept at the top, just that `dist` is now a multi-line value-setting.

Reply:
The change to use a function simplifies it enough for me. Just need to fix it so it returns the value...